packages/ui/src/components/Visualization/Canvas/CanvasForm.tsx

```diff
@@ -38,12 +39,16 @@ export const CanvasForm: FunctionComponent<CanvasFormProps> = (props) => {
     [entitiesContext, props.selectedNode.data?.vizNode],
   );
 
+  const isExpressionAwareStep = useMemo(() => {
+    return schema?.schema?.properties?.expression !== undefined;
+  }, [schema]);
+
   return schema?.schema === undefined ? null : (
     <ErrorBoundary fallback={<p>This node cannot be configured yet</p>}>
       <Title headingLevel="h1">{componentName}</Title>
-
+      {isExpressionAwareStep && <ExpressionEditor selectedNode={props.selectedNode} />}
```

Comment:
1. Since this is done outside of the uniforms, `CustomAutoField`  doesn't participate
2. Can you ellaborate?
3. I'd like to wait a bit to see how `Integration`  and `Kamelet` kind is implemented, as this logic will be used for those as well

Reply:
1. Could it be incorporated into the `CustomAutoField` workflow? This is not a blocking issue in any case
2. I meant what would happen if a `Pipe` for instance has an `expression` property? I guess we're gonna see the expression field anyway.
3. That sounds fair :+1: